R/cast.R

```diff
@@ -0,0 +1,21 @@
+up_cast <- function(object, class) {
+  check_R7(object)
+
+  class <- as_class(class)
+  check_can_inherit(class)
+  if (!class_inherits(object, class)) {
+    msg <- sprintf("`object` %s does not inherit from %s", obj_desc(object), class_desc(class))
+    stop(msg)
+  }
+
+  # Must not change order of these fields as C code indexes by position
+  structure(
+    list(
+      object = object,
+      dispatch = class_register(class)
```

Comment:
We want to force dispatch on the register class, because that’s the terminal class name. It’s a bit surprising but if you walk through `class_dispatch` vs `class_register` I think you should be able to convince yourself it’s correct. 

But maybe `up_cast` isn’t quite the right name? The goal is to force dispatch to a specific class, disallowing further inheritance (since that’s one of the things that makes next method hard to reason about).